# Análisis de la Calidad del Aire en el Valle de México

## Introducción

La Organización Mundial de la Salud estima que en 2012 aconteció un aproximado de 3.7 millones de muertes prematuras derivadas de enfermedades atribuidas a la contaminación del aire. Esto se refleja en su discurso del 2014: “La contaminación del aire es el riesgo para la salud ambiental número uno del mundo”.

  

La Ciudad de México, centro político, histórico y financiero de nuestro país cuenta con poco más de 9 millones de habitantes convirtiéndose en la ciudad más poblada de norteamérica, con una densidad de 600,000 habitantes por kilómetro cuadrado. Esto sin contar la totalidad de población que habita la Zona Metropolitana del Valle de México, que se contabiliza por encima de los 21 millones de habitantes.

  

Los problemas que aquejan a la incomparable urbanización no son pocos, desde crimen organizado, alcoholismo, inundaciones periódicas, fríos intensos y el que concierne a esta investigación, la contaminación del aire. De las majestuosas vistas que los primeros españoles hallaron en aquel valle, no queda nada. Incluso el propio cielo se esconde entre una densa nube de vapores tóxicos.

  

Siendo conscientes de este problema, desde el año de 1986, se monitorea de forma periódica la calidad del aire en la Zona Metropolitana del Valle de México por medio del Sistema de Monitoreo Atmosférico (SIMAT) y la RAMA (Red Automática de Monitoreo Atmosférico), realizando mediciones de la concentración de los contaminantes criterio cada hora.

  

Ahora bien, los contaminantes criterio son aquellos que afectan el bienestar y la salud humana, por lo que cuentan con criterios para establecer o revisar límites máximos permisibles por medio de las Normas Oficiales Mexicanas. La concentración de estos contaminantes sirve para indicar la calidad del aire.

  

Dentro del grupo de contaminantes criterio se encuentran el dióxido de azufre (SO2), el dióxido de nitrógeno (NO2), el monóxido de carbono (CO), el ozono (O3) y las partículas en suspensión (PM10 y PM2.5).

Estos contaminantes tienen graves afectaciones a la salud por lo que es necesario no solo monitorear su concentración, sino también obtener con ellos un índice que nos permita clasificar la calidad del aire para así comunicarlo a la población en general. Con este objetivo en mente, en 1982 se diseñó el Índice Metropolitano de la Calidad del Aire (IMECA), cuya metodología transforma a una escala adimensional las concentraciones de los contaminantes criterio; así podemos entender que tan buena o mala es la calidad del aire observando el valor IMECA de dicho contaminante e incluso compararlo con otro, ya que tienen la misma escala.

![](https://lh5.googleusercontent.com/07HClik8fpzXOVvVHCpbBIsc_ku2rtnSCv2p07o0JHnHgUkJp02HyDIQXY0dt8j-ISiy0Rh-wJ-YaKGbYCaPjuk2qQiNh-puh-JleOWb-Fj4z-xp9CrghqKhtReMHtEGzXDHun8a)

                                              Clasificación IMECA
                                              
Durante nuestra investigación nos dimos cuenta que la información sobre la calidad del aire en la Zona Metropolitana del Valle de México no es tan accesible: existen los datos y están abiertos al público, sin embargo, hay muy poca difusión sobre sus resultados, el último informe anual de la calidad del aire en ciudad de méxico se realizó en el 2018. Por lo tanto el objetivo de este proyecto es primero que nada analizar los datos disponibles para entender la problemática de la contaminación del aire en el Valle de México, y hacer que esta información sea más accesible, para luego proponer posibles soluciones.

## Preguntas de Investigación

Planteamos las siguientes preguntas para ayudar a entender y resolver el problema planteado:

-   ¿Cuál es el promedio histórico (considerando los años 2005-2020) de concentración de cada contaminante por zona?

    
-   ¿En qué meses se tiene la máxima concentración de cada contaminante?

    
-   ¿Hay algún mes de algún año en el que se haya obtenido la calificación de calidad “Extremadamente mala”?

    
-   ¿Cuál es la zona que presenta mayor contaminación, en general?

    
-   ¿Cuál es el contaminante con más concentración por año (del 2005- 2020)?

    
-   ¿Cuál fue y cuándo se registró la concentración más baja durante 2020?

    
-   ¿Cuál fue y cuándo se registró la mayor concentración durante el periodo de confinamiento?

    
-   ¿El periodo de confinamiento coincide con una mejor calidad de aire?

    
-   ¿Cuál fue el contaminante con menores cambios en el período 2005-2020?

## Módulo: Procesamiento de Datos

### Lectura y Análisis exploratorio de los datos

In [1]:
#Importando paquetes
import pandas as pd
import numpy as np

In [2]:
#Leyendo un archivo de datos IMECA para averiguar que contiene
df=pd.read_csv("datos_IMECA/csv/imeca2005.csv")
df

Fecha  Hora  Noroeste Ozono  Noroeste dioxido de azufre  \
0     01/01/2005     1              19                          19   
1     01/01/2005     2              19                          19   
2     01/01/2005     3              17                          19   
3     01/01/2005     4              16                          19   
4     01/01/2005     5              16                          19   
...          ...   ...             ...                         ...   
8755  31/12/2005    20              15                          10   
8756  31/12/2005    21              12                          10   
8757  31/12/2005    22               9                          10   
8758  31/12/2005    23               8                          11   
8759  31/12/2005    24               9                          11   

      Noroeste dioxido de nitrogeno  Noroeste monoxido de carbono  \
0                                43                            42   
1                                42                            45   
2                                42                            48   
3                                42                            52   
4                                39                            53   
...                             ...                           ...   
8755                             38                            18   
8756                             34                            20   
8757                             31                            23   
8758                             25                            27   
8759                             23                            30   

      Noroeste PM10  Noreste Ozono  Noreste dioxido de azufre  \
0                80             10                         22   
1                84              7                         22   
2                88              5                         22   
3                91              6                         22   
4                94              8                         23   
...             ...            ...                        ...   
8755             61             41                         12   
8756             61             35                         12   
8757             61             28                         12   
8758             62             23                         12   
8759             64             22                         12   

      Noreste dioxido de nitrogeno  ...  Suroeste Ozono  \
0                               34  ...              26   
1                               35  ...              19   
2                               34  ...              23   
3                               33  ...              21   
4                               34  ...              23   
...                            ...  ...             ...   
8755                            34  ...              33   
8756                            29  ...              27   
8757                            33  ...              19   
8758                            31  ...              22   
8759                            31  ...              20   

      Suroeste dioxido de azufre  Suroeste dioxido de nitrogeno  \
0                             12                             39   
1                             12                             34   
2                             12                             33   
3                             12                             32   
4                             12                             31   
...                          ...                            ...   
8755                           5                             24   
8756                           5                             30   
8757                           5                             30   
8758                           5                             25   
8759                           5                             24   

      Suroeste monoxido de carbono  Suroeste P

In [3]:
#Revisando dimensión
df.shape

(8760, 27)

In [4]:
#Revisando columnas que contiene
df.columns

Index(['Fecha', 'Hora', 'Noroeste Ozono', 'Noroeste dioxido de azufre',
       'Noroeste dioxido de nitrogeno', 'Noroeste monoxido de carbono',
       'Noroeste PM10', 'Noreste Ozono', 'Noreste dioxido de azufre',
       'Noreste dioxido de nitrogeno', 'Noreste monoxido de carbono',
       'Noreste PM10', 'Centro Ozono', 'Centro dioxido de azufre',
       'Centro dioxido de nitrogeno', 'Centro monoxido de carbono',
       'Centro PM10', 'Suroeste Ozono', 'Suroeste dioxido de azufre',
       'Suroeste dioxido de nitrogeno', 'Suroeste monoxido de carbono',
       'Suroeste PM10', 'Sureste Ozono', 'Sureste dioxido de azufre',
       'Sureste dioxido de nitrogeno', 'Sureste monoxido de carbono',
       'Sureste PM10'],
      dtype='object')

Tenemos un archivo de datos por año, vamos a unir todos en un solo DataFrame para trabajar con ellos más facilmente

In [5]:
#primero creamos una lista que contenga todas las rutas de acceso a los archivos csv
lista=[ f'datos_IMECA/csv/imeca{2005+i}.csv' for i in range(0,16)]
lista

['datos_IMECA/csv/imeca2005.csv',
 'datos_IMECA/csv/imeca2006.csv',
 'datos_IMECA/csv/imeca2007.csv',
 'datos_IMECA/csv/imeca2008.csv',
 'datos_IMECA/csv/imeca2009.csv',
 'datos_IMECA/csv/imeca2010.csv',
 'datos_IMECA/csv/imeca2011.csv',
 'datos_IMECA/csv/imeca2012.csv',
 'datos_IMECA/csv/imeca2013.csv',
 'datos_IMECA/csv/imeca2014.csv',
 'datos_IMECA/csv/imeca2015.csv',
 'datos_IMECA/csv/imeca2016.csv',
 'datos_IMECA/csv/imeca2017.csv',
 'datos_IMECA/csv/imeca2018.csv',
 'datos_IMECA/csv/imeca2019.csv',
 'datos_IMECA/csv/imeca2020.csv']

In [6]:
#Realizamos la lectura de todos los archivos csv juntos y reemplazamos -99 por NaN
DF_list = [(pd.read_csv(f'{i}',sep=',')).replace(-99, np.nan) for i in lista]

In [7]:
#comprobamos que se puedan leer de forma correcta los datos
DF_list[0]

Fecha  Hora  Noroeste Ozono  Noroeste dioxido de azufre  \
0     01/01/2005     1              19                          19   
1     01/01/2005     2              19                          19   
2     01/01/2005     3              17                          19   
3     01/01/2005     4              16                          19   
4     01/01/2005     5              16                          19   
...          ...   ...             ...                         ...   
8755  31/12/2005    20              15                          10   
8756  31/12/2005    21              12                          10   
8757  31/12/2005    22               9                          10   
8758  31/12/2005    23               8                          11   
8759  31/12/2005    24               9                          11   

      Noroeste dioxido de nitrogeno  Noroeste monoxido de carbono  \
0                                43                            42   
1                                42                            45   
2                                42                            48   
3                                42                            52   
4                                39                            53   
...                             ...                           ...   
8755                             38                            18   
8756                             34                            20   
8757                             31                            23   
8758                             25                            27   
8759                             23                            30   

      Noroeste PM10  Noreste Ozono  Noreste dioxido de azufre  \
0                80             10                         22   
1                84              7                         22   
2                88              5                         22   
3                91              6                         22   
4                94              8                         23   
...             ...            ...                        ...   
8755             61             41                         12   
8756             61             35                         12   
8757             61             28                         12   
8758             62             23                         12   
8759             64             22                         12   

      Noreste dioxido de nitrogeno  ...  Suroeste Ozono  \
0                               34  ...              26   
1                               35  ...              19   
2                               34  ...              23   
3                               33  ...              21   
4                               34  ...              23   
...                            ...  ...             ...   
8755                            34  ...              33   
8756                            29  ...              27   
8757                            33  ...              19   
8758                            31  ...              22   
8759                            31  ...              20   

      Suroeste dioxido de azufre  Suroeste dioxido de nitrogeno  \
0                             12                             39   
1                             12                             34   
2                             12                             33   
3                             12                             32   
4                             12                             31   
...                          ...                            ...   
8755                           5                             24   
8756                           5                             30   
8757                           5                             30   
8758                           5                             25   
8759                           5                             24   

      Suroeste monoxido de carbono  Suroeste P

In [8]:
#segunda comprobación
DF_list[13]

Fecha  Hora  Noroeste Ozono  Noroeste dioxido de azufre  \
0     01/01/2018     1            24.0                           8   
1     01/01/2018     2            23.0                           8   
2     01/01/2018     3            18.0                           8   
3     01/01/2018     4            12.0                          10   
4     01/01/2018     5            11.0                          12   
...          ...   ...             ...                         ...   
8755  31/12/2018    20            20.0                           5   
8756  31/12/2018    21            12.0                           5   
8757  31/12/2018    22             6.0                           5   
8758  31/12/2018    23             4.0                           5   
8759  31/12/2018    24             NaN                           5   

      Noroeste dioxido de nitrogeno  Noroeste monoxido de carbono  \
0                              14.0                          12.0   
1                              18.0                          12.0   
2                              19.0                          12.0   
3                              18.0                          12.0   
4                              18.0                          12.0   
...                             ...                           ...   
8755                           26.0                           5.0   
8756                           26.0                           6.0   
8757                           26.0                           7.0   
8758                           22.0                           8.0   
8759                            NaN                          12.0   

      Noroeste PM10  Noreste Ozono  Noreste dioxido de azufre  \
0               106           24.0                          5   
1               107           29.0                          5   
2               108           23.0                          5   
3               110           18.0                          5   
4               113           17.0                          5   
...             ...            ...                        ...   
8755             46           17.0                         12   
8756             49           14.0                         12   
8757             50           11.0                         12   
8758             54            6.0                         12   
8759             61            NaN                         12   

      Noreste dioxido de nitrogeno  ...  Suroeste dioxido de nitrogeno  \
0                             18.0  ...                           17.0   
1                             16.0  ...                           12.0   
2                             19.0  ...                           10.0   
3                             17.0  ...                            9.0   
4                             18.0  ...                            9.0   
...                            ...  ...                            ...   
8755                          29.0  ...                           18.0   
8756                          20.0  ...                           25.0   
8757                          21.0  ...                           26.0   
8758                          21.0  ...                           21.0   
8759                           NaN  ...                            NaN   

      Suroeste monoxido de carbono  Suroeste PM10  Sureste Ozono  \
0                              8.0             67            5.0   
1                              7.0             70            8.0   
2                              7.0             73            4.0   
3                              7.0             76            6.0   
4                              6.0             77            4.0   
...                            ...            ...            ...   
8755                           4.0             26           16.0   
8756                           5.0             28           11.0   
8757                           6.0             30          

In [9]:
#El Dataframe número 14 y 15 tienen 32 columnas cuando lo correcto serían 27
#Eliminamos las columnas que sobran
cols = [7,13,19,25,31]
DF_list[14]=DF_list[14].drop(DF_list[14].columns[cols],axis=1)
DF_list[15]=DF_list[15].drop(DF_list[15].columns[cols],axis=1)

In [10]:
#Se puede observar que existe un inconveniente: por alguna razón algunos DataFrames 
#tienen columnas de más, que contienen solo elementos vacíos, es necesario eliminarlas
DF_list[12].columns

Index(['Fecha', 'Hora', 'Noroeste Ozono', 'Noroeste dioxido de azufre',
       'Noroeste dioxido de nitrogeno', 'Noroeste monoxido de carbono',
       'Noroeste PM10', 'Noreste Ozono', 'Noreste dioxido de azufre',
       'Noreste dioxido de nitrogeno', 'Noreste monoxido de carbono',
       'Noreste PM10', 'Centro Ozono', 'Centro dioxido de azufre',
       'Centro dioxido de nitrogeno', 'Centro monoxido de carbono',
       'Centro PM10', 'Suroeste Ozono', 'Suroeste dioxido de azufre',
       'Suroeste dioxido de nitrogeno', 'Suroeste monoxido de carbono',
       'Suroeste PM10', 'Sureste Ozono', 'Sureste dioxido de azufre',
       'Sureste dioxido de nitrogeno', 'Sureste monoxido de carbono',
       'Sureste PM10', 'Unnamed: 27', 'Unnamed: 28'],
      dtype='object')

In [11]:
DF_list[12].shape

(8760, 29)

In [12]:
DF_list[12]['Unnamed: 27'] #Se observa que contiene puros elementos vacíos

0      NaN
1      NaN
2      NaN
3      NaN
4      NaN
        ..
8755   NaN
8756   NaN
8757   NaN
8758   NaN
8759   NaN
Name: Unnamed: 27, Length: 8760, dtype: float64

In [13]:
#ahora eliminamos las columnas que aparecen como sobrantes
DF_list2=[i.dropna(axis="columns", how='all') for i in DF_list]
DF_list2[0]

Fecha  Hora  Noroeste Ozono  Noroeste dioxido de azufre  \
0     01/01/2005     1              19                          19   
1     01/01/2005     2              19                          19   
2     01/01/2005     3              17                          19   
3     01/01/2005     4              16                          19   
4     01/01/2005     5              16                          19   
...          ...   ...             ...                         ...   
8755  31/12/2005    20              15                          10   
8756  31/12/2005    21              12                          10   
8757  31/12/2005    22               9                          10   
8758  31/12/2005    23               8                          11   
8759  31/12/2005    24               9                          11   

      Noroeste dioxido de nitrogeno  Noroeste monoxido de carbono  \
0                                43                            42   
1                                42                            45   
2                                42                            48   
3                                42                            52   
4                                39                            53   
...                             ...                           ...   
8755                             38                            18   
8756                             34                            20   
8757                             31                            23   
8758                             25                            27   
8759                             23                            30   

      Noroeste PM10  Noreste Ozono  Noreste dioxido de azufre  \
0                80             10                         22   
1                84              7                         22   
2                88              5                         22   
3                91              6                         22   
4                94              8                         23   
...             ...            ...                        ...   
8755             61             41                         12   
8756             61             35                         12   
8757             61             28                         12   
8758             62             23                         12   
8759             64             22                         12   

      Noreste dioxido de nitrogeno  ...  Suroeste Ozono  \
0                               34  ...              26   
1                               35  ...              19   
2                               34  ...              23   
3                               33  ...              21   
4                               34  ...              23   
...                            ...  ...             ...   
8755                            34  ...              33   
8756                            29  ...              27   
8757                            33  ...              19   
8758                            31  ...              22   
8759                            31  ...              20   

      Suroeste dioxido de azufre  Suroeste dioxido de nitrogeno  \
0                             12                             39   
1                             12                             34   
2                             12                             33   
3                             12                             32   
4                             12                             31   
...                          ...                            ...   
8755                           5                             24   
8756                           5                             30   
8757                           5                             30   
8758                           5                             25   
8759                           5                             24   

      Suroeste monoxido de carbono  Suroeste P

In [14]:
#observamos que las columnas sobrantes se fueron
DF_list2[12].columns

Index(['Fecha', 'Hora', 'Noroeste Ozono', 'Noroeste dioxido de azufre',
       'Noroeste dioxido de nitrogeno', 'Noroeste monoxido de carbono',
       'Noroeste PM10', 'Noreste Ozono', 'Noreste dioxido de azufre',
       'Noreste dioxido de nitrogeno', 'Noreste monoxido de carbono',
       'Noreste PM10', 'Centro Ozono', 'Centro dioxido de azufre',
       'Centro dioxido de nitrogeno', 'Centro monoxido de carbono',
       'Centro PM10', 'Suroeste Ozono', 'Suroeste dioxido de azufre',
       'Suroeste dioxido de nitrogeno', 'Suroeste monoxido de carbono',
       'Suroeste PM10', 'Sureste Ozono', 'Sureste dioxido de azufre',
       'Sureste dioxido de nitrogeno', 'Sureste monoxido de carbono',
       'Sureste PM10'],
      dtype='object')

In [15]:
#Revisamos que todos los DataFrame tengan el mismo tamaño
for i in range(0,16):
    print(np.shape(DF_list2[i]))

(8760, 27)
(8760, 27)
(8760, 27)
(8784, 27)
(8760, 27)
(8760, 27)
(8760, 27)
(8784, 27)
(8781, 27)
(8760, 27)
(8760, 27)
(8784, 27)
(8760, 27)
(8760, 27)
(8760, 27)
(8784, 27)


Se observa que hay años en los que se tiene 8784 filas en lugar de 8760, esto es porque se trata de años bisiestos y tienen un día más, lo que equivale a 24 observaciones extra.

In [16]:
#Renombramos todas las columnas para no generar problemas al momento de concatenar
#DF_list2[0].columns

nombres_columnas=( "fecha"           ,        "hora"                ,          "noroeste_ozono"               ,
 "noroeste_dioxido_de_azufre"   , "noroeste_dioxido_de_nitrogeno", "noroeste_monoxido_de_carbono" ,
 "noroeste_pm10"                , "noreste_ozono"                , "noreste_dioxido_de_azufre"    ,
 "noreste_dioxido_de_nitrogeno" , "noreste_monoxido_de_carbono"  , "noreste_pm10"                 ,
 "centro_ozono"                 , "centro_dioxido_de_azufre"     , "centro_dioxido_de_nitrogeno"  ,
 "centro_monoxido_de_carbono"   , "centro_pm10"                  , "suroeste_ozono"               ,
 "suroeste_dioxido_de_azufre"   , "suroeste_dioxido_de_nitrogeno", "suroeste_monoxido_de_carbono" ,
 "suroeste_pm10"                , "sureste_ozono"                , "sureste_dioxido_de_azufre"    ,
 "sureste_dioxido_de_nitrogeno" , "sureste_monoxido_de_carbono"  , "sureste_pm10" )

for i in DF_list2:
    i.columns = nombres_columnas

In [17]:
#Concatenando todo en un mismo DataFrame
DF_all = pd.concat(DF_list2,axis=0)
DF_all=DF_all.reset_index(drop=True)
DF_all

fecha  hora  noroeste_ozono  noroeste_dioxido_de_azufre  \
0       01/01/2005   1.0            19.0                        19.0   
1       01/01/2005   2.0            19.0                        19.0   
2       01/01/2005   3.0            17.0                        19.0   
3       01/01/2005   4.0            16.0                        19.0   
4       01/01/2005   5.0            16.0                        19.0   
...            ...   ...             ...                         ...   
140272  31/12/2020  20.0            25.0                         4.0   
140273  31/12/2020  21.0            22.0                         4.0   
140274  31/12/2020  22.0            18.0                         4.0   
140275  31/12/2020  23.0            16.0                         4.0   
140276  31/12/2020  24.0            16.0                         4.0   

        noroeste_dioxido_de_nitrogeno  noroeste_monoxido_de_carbono  \
0                                43.0                          42.0   
1                                42.0                          45.0   
2                                42.0                          48.0   
3                                42.0                          52.0   
4                                39.0                          53.0   
...                               ...                           ...   
140272                            9.0                           2.0   
140273                            9.0                           2.0   
140274                           18.0                           3.0   
140275                           18.0                           5.0   
140276                           18.0                           5.0   

        noroeste_pm10  noreste_ozono  noreste_dioxido_de_azufre  \
0                80.0           10.0                       22.0   
1                84.0            7.0                       22.0   
2                88.0            5.0                       22.0   
3                91.0            6.0                       22.0   
4                94.0            8.0                       23.0   
...               ...            ...                        ...   
140272           88.0           14.0                        4.0   
140273           88.0           11.0                        4.0   
140274           88.0           10.0                        4.0   
140275           88.0            9.0                        4.0   
140276           88.0           11.0                        4.0   

        noreste_dioxido_de_nitrogeno  ...  suroeste_ozono  \
0                               34.0  ...            26.0   
1                               35.0  ...            19.0   
2                               34.0  ...            23.0   
3                               33.0  ...            21.0   
4                               34.0  ...            23.0   
...                              ...  ...             ...   
140272                          15.0  ...            25.0   
140273                          17.0  ...            16.0   
140274                          17.0  ...             9.0   
140275                          19.0  ...            11.0   
140276                          18.0  ...            16.0   

        suroeste_dioxido_de_azufre  suroeste_dioxido_de_nitrogeno  \
0                             12.0                           39.0   
1                             12.0                           34.0   
2                             12.0                           33.0   
3                             12.0                           32.0   
4                             12.0                           31.0   
...                            ...                            ...   
140272                         2.0                           14.0   
140273                         2.0                           15.0   
140274                         2.0                           16.0   
140275                         2.0                           15.0  

### Limpieza NaNs

In [18]:
#Revisando datos faltantes por columna
DF_all.isna().sum()

fecha                              21
hora                               21
noroeste_ozono                   1306
noroeste_dioxido_de_azufre         22
noroeste_dioxido_de_nitrogeno    1297
noroeste_monoxido_de_carbono     2027
noroeste_pm10                     174
noreste_ozono                    1299
noreste_dioxido_de_azufre          21
noreste_dioxido_de_nitrogeno     1331
noreste_monoxido_de_carbono      2073
noreste_pm10                      369
centro_ozono                     1319
centro_dioxido_de_azufre          101
centro_dioxido_de_nitrogeno      1366
centro_monoxido_de_carbono       2070
centro_pm10                       555
suroeste_ozono                   1236
suroeste_dioxido_de_azufre         42
suroeste_dioxido_de_nitrogeno    1261
suroeste_monoxido_de_carbono     2050
suroeste_pm10                     365
sureste_ozono                    1319
sureste_dioxido_de_azufre         126
sureste_dioxido_de_nitrogeno     1485
sureste_monoxido_de_carbono      2231
sureste_pm10

Se nota que hay columnas que presentan más datos faltantes que otras. Puesto que no podemos despreciar las mediciones de toda la fila solo porque una observación en una columna es nula, optamos en su lugar por llenar los valores vacíos con el valor promedio de la columna. De esta forma nos quedamos con suficientes datos para trabajar y mantenemos su valor promedio.

In [19]:
#Llenando los valores numéricos con el valor promedio
DF_all_subset=DF_all.loc[:,"noroeste_ozono":"sureste_pm10"]

for i in DF_all_subset.columns:
    DF_all[i]=DF_all[i].fillna(DF_all[i].mean())

In [20]:
#Corroborando que se hayan reemplazado todos los datos faltantes numéricos
DF_all.isna().sum()

fecha                            21
hora                             21
noroeste_ozono                    0
noroeste_dioxido_de_azufre        0
noroeste_dioxido_de_nitrogeno     0
noroeste_monoxido_de_carbono      0
noroeste_pm10                     0
noreste_ozono                     0
noreste_dioxido_de_azufre         0
noreste_dioxido_de_nitrogeno      0
noreste_monoxido_de_carbono       0
noreste_pm10                      0
centro_ozono                      0
centro_dioxido_de_azufre          0
centro_dioxido_de_nitrogeno       0
centro_monoxido_de_carbono        0
centro_pm10                       0
suroeste_ozono                    0
suroeste_dioxido_de_azufre        0
suroeste_dioxido_de_nitrogeno     0
suroeste_monoxido_de_carbono      0
suroeste_pm10                     0
sureste_ozono                     0
sureste_dioxido_de_azufre         0
sureste_dioxido_de_nitrogeno      0
sureste_monoxido_de_carbono       0
sureste_pm10                      0
dtype: int64

In [21]:
#Revisando valores nulos de fechas
DF_all[DF_all["fecha"].isna()]

fecha  hora  noroeste_ozono  noroeste_dioxido_de_azufre  \
78888   NaN   NaN       28.903253                    9.383715   
78889   NaN   NaN       28.903253                    9.383715   
78890   NaN   NaN       28.903253                    9.383715   
78891   NaN   NaN       28.903253                    9.383715   
78892   NaN   NaN       28.903253                    9.383715   
78893   NaN   NaN       28.903253                    9.383715   
78894   NaN   NaN       28.903253                    9.383715   
78895   NaN   NaN       28.903253                    9.383715   
78896   NaN   NaN       28.903253                    9.383715   
78897   NaN   NaN       28.903253                    9.383715   
78898   NaN   NaN       28.903253                    9.383715   
78899   NaN   NaN       28.903253                    9.383715   
78900   NaN   NaN       28.903253                    9.383715   
78901   NaN   NaN       28.903253                    9.383715   
78902   NaN   NaN       28.903253                    9.383715   
78903   NaN   NaN       28.903253                    9.383715   
78904   NaN   NaN       28.903253                    9.383715   
78905   NaN   NaN       28.903253                    9.383715   
78906   NaN   NaN       28.903253                    9.383715   
78907   NaN   NaN       28.903253                    9.383715   
78908   NaN   NaN       28.903253                    9.383715   

       noroeste_dioxido_de_nitrogeno  noroeste_monoxido_de_carbono  \
78888                      17.478565                     10.089902   
78889                      17.478565                     10.089902   
78890                      17.478565                     10.089902   
78891                      17.478565                     10.089902   
78892                      17.478565                     10.089902   
78893                      17.478565                     10.089902   
78894                      17.478565                     10.089902   
78895                      17.478565                     10.089902   
78896                      17.478565                     10.089902   
78897                      17.478565                     10.089902   
78898                      17.478565                     10.089902   
78899                      17.478565                     10.089902   
78900                      17.478565                     10.089902   
78901                      17.478565                     10.089902   
78902                      17.478565                     10.089902   
78903                      17.478565                     10.089902   
78904                      17.478565                     10.089902   
78905                      17.478565                     10.089902   
78906                      17.478565                     10.089902   
78907                      17.478565                     10.089902   
78908                      17.478565                     10.089902   

       noroeste_pm10  noreste_ozono  noreste_dioxido_de_azufre  \
78888      55.736244      31.219862                   7.527464   
78889      55.736244      31.219862                   7.527464   
78890      55.736244      31.219862                   7.527464   
78891      55.736244      31.219862                   7.527464   
78892      55.736244      31.219862                   7.527464   
78893      55.736244      31.219862                   7.527464   
78894      55.736244      31.219862                   7.527464   
78895      55.736244      31.219862                   7.527464   
78896      55.736244      31.219862                   7.527464   
78897      55.736244      31.219862                   7.527464   
78898      55.736244      31.219862                   7.527464   
78899      55.736244      31.219862                   7.527464   
78900      55.736244      31.219862                   7.527464   
78901      55.736244      31.219862                   7.527464   
78902      55.736244      31.219862         

In [22]:
#Revisando la última fecha correcta
DF_all.loc[78887]

fecha                            31/12/2013
hora                                     24
noroeste_ozono                            9
noroeste_dioxido_de_azufre               22
noroeste_dioxido_de_nitrogeno            15
noroeste_monoxido_de_carbono              7
noroeste_pm10                            48
noreste_ozono                             7
noreste_dioxido_de_azufre                11
noreste_dioxido_de_nitrogeno             15
noreste_monoxido_de_carbono              12
noreste_pm10                             48
centro_ozono                              1
centro_dioxido_de_azufre                 16
centro_dioxido_de_nitrogeno              16
centro_monoxido_de_carbono                9
centro_pm10                              48
suroeste_ozono                            5
suroeste_dioxido_de_azufre               15
suroeste_dioxido_de_nitrogeno            20
suroeste_monoxido_de_carbono             13
suroeste_pm10                            51
sureste_ozono                   

In [23]:
#Revisando fecha posterior a la última vacía
DF_all.loc[78909]

fecha                            01/01/2014
hora                                      1
noroeste_ozono                            8
noroeste_dioxido_de_azufre               18
noroeste_dioxido_de_nitrogeno            15
noroeste_monoxido_de_carbono              7
noroeste_pm10                            51
noreste_ozono                             8
noreste_dioxido_de_azufre                11
noreste_dioxido_de_nitrogeno             15
noreste_monoxido_de_carbono              13
noreste_pm10                             48
centro_ozono                              1
centro_dioxido_de_azufre                 15
centro_dioxido_de_nitrogeno              15
centro_monoxido_de_carbono                9
centro_pm10                              48
suroeste_ozono                            4
suroeste_dioxido_de_azufre               14
suroeste_dioxido_de_nitrogeno            18
suroeste_monoxido_de_carbono             14
suroeste_pm10                            53
sureste_ozono                   

In [24]:
#Se corrobora que el archivo de 2014 esté correcto
pd.read_csv("datos_IMECA/csv/imeca2014.csv")

Fecha  Hora  Noroeste Ozono  Noroeste dioxido de azufre  \
0     01/01/2014     1               8                          18   
1     01/01/2014     2               8                          18   
2     01/01/2014     3               5                          18   
3     01/01/2014     4               2                          18   
4     01/01/2014     5               3                          18   
...          ...   ...             ...                         ...   
8755  31/12/2014    20              27                          25   
8756  31/12/2014    21              18                          25   
8757  31/12/2014    22              12                          25   
8758  31/12/2014    23               9                          26   
8759  31/12/2014    24              12                          26   

      Noroeste dioxido de nitrogeno  Noroeste monoxido de carbono  \
0                                15                             7   
1                                14                             7   
2                                14                             7   
3                                14                             9   
4                                13                            11   
...                             ...                           ...   
8755                             31                            13   
8756                             29                            12   
8757                             30                            12   
8758                             27                            13   
8759                             25                            14   

      Noroeste PM10  Noreste Ozono  Noreste dioxido de azufre  \
0                51              8                         11   
1                52              8                         12   
2                53              8                         13   
3                53              7                         14   
4                56              5                         15   
...             ...            ...                        ...   
8755            111             26                         34   
8756            111             13                         34   
8757            111              6                         34   
8758            111              4                         35   
8759            111              3                         34   

      Noreste dioxido de nitrogeno  ...  Suroeste dioxido de nitrogeno  \
0                               15  ...                             18   
1                               14  ...                             16   
2                               13  ...                             15   
3                               13  ...                             14   
4                               12  ...                             13   
...                            ...  ...                            ...   
8755                            44  ...                             61   
8756                            32  ...                             49   
8757                            29  ...                             49   
8758                            23  ...                             45   
8759                            19  ...                             34   

      Suroeste monoxido de carbono  Suroeste PM10  Sureste Ozono  \
0                               14             53              3   
1                               14             53              5   
2                               13             53              4   
3                               13             54              4   
4                               14             56              4   
...                            ...            ...            ...   
8755                            19            116             76   
8756                            20            116             24   
8757                            22            117          

In [25]:
#Observando el archivo del 2013
pd.read_csv("datos_IMECA/csv/imeca2013.csv")

Fecha  Hora  Noroeste Ozono  Noroeste dioxido de azufre  \
0     01/01/2013   1.0             3.0                         2.0   
1     01/01/2013   2.0             3.0                         3.0   
2     01/01/2013   3.0             4.0                         3.0   
3     01/01/2013   4.0             5.0                         3.0   
4     01/01/2013   5.0             9.0                         4.0   
...          ...   ...             ...                         ...   
8776         NaN   NaN             NaN                         NaN   
8777         NaN   NaN             NaN                         NaN   
8778         NaN   NaN             NaN                         NaN   
8779         NaN   NaN             NaN                         NaN   
8780         NaN   NaN             NaN                         NaN   

      Noroeste dioxido de nitrogeno  Noroeste monoxido de carbono  \
0                              17.0                          10.0   
1                              16.0                          12.0   
2                              16.0                          15.0   
3                              15.0                          17.0   
4                              16.0                          20.0   
...                             ...                           ...   
8776                            NaN                           NaN   
8777                            NaN                           NaN   
8778                            NaN                           NaN   
8779                            NaN                           NaN   
8780                            NaN                           NaN   

      Noroeste PM10  Noreste Ozono  Noreste dioxido de azufre  \
0              37.0           11.0                        4.0   
1              42.0            7.0                        5.0   
2              48.0            7.0                        5.0   
3              52.0            7.0                        5.0   
4              57.0            6.0                        5.0   
...             ...            ...                        ...   
8776            NaN            NaN                        NaN   
8777            NaN            NaN                        NaN   
8778            NaN            NaN                        NaN   
8779            NaN            NaN                        NaN   
8780            NaN            NaN                        NaN   

      Noreste dioxido de nitrogeno  ...  Suroeste PM10  Sureste Ozono  \
0                             20.0  ...           23.0            3.0   
1                             18.0  ...           28.0            7.0   
2                             19.0  ...           33.0            4.0   
3                             18.0  ...           38.0            4.0   
4                             17.0  ...           41.0            5.0   
...                            ...  ...            ...            ...   
8776                           NaN  ...            NaN            NaN   
8777                           NaN  ...            NaN            NaN   
8778                           NaN  ...            NaN            NaN   
8779                           NaN  ...            NaN            NaN   
8780                           NaN  ...            NaN            NaN   

      Sureste dioxido de azufre  Sureste dioxido de nitrogeno  \
0                           3.0                          18.0   
1                           3.0                          16.0   
2                           3.0                          16.0   
3                           3.0                          16.0   
4                           3.0                          15.0   
...                         ...                           ...   
8776                        NaN                           NaN   
8777                        NaN                           NaN   
8778                        NaN                           NaN   
8779                        NaN  

Como se puede observar el archivo del 2013 contiene 8781 filas cuando debería contener 8760 ya que no es bisiesto. Es decir, tiene 21 filas vacías extra. Es necesario eliminar estas filas extra que no aportan información, esto se hará desde DF_all (el que acabamos de limpiar y contiene todos los arvhivos), puesto que son los únicos elementos vacíos que quedan.

In [26]:
#Tamaño antes de limpiar
DF_all.shape

(140277, 27)

In [27]:
#Eliminando filas con fecha vacía
DF_all=DF_all.dropna(axis="rows")

In [28]:
#Tamaño después de limpiar
DF_all.shape

(140256, 27)

In [29]:
#Corroborando que no queden elementos nulos
DF_all.isna().sum()

fecha                            0
hora                             0
noroeste_ozono                   0
noroeste_dioxido_de_azufre       0
noroeste_dioxido_de_nitrogeno    0
noroeste_monoxido_de_carbono     0
noroeste_pm10                    0
noreste_ozono                    0
noreste_dioxido_de_azufre        0
noreste_dioxido_de_nitrogeno     0
noreste_monoxido_de_carbono      0
noreste_pm10                     0
centro_ozono                     0
centro_dioxido_de_azufre         0
centro_dioxido_de_nitrogeno      0
centro_monoxido_de_carbono       0
centro_pm10                      0
suroeste_ozono                   0
suroeste_dioxido_de_azufre       0
suroeste_dioxido_de_nitrogeno    0
suroeste_monoxido_de_carbono     0
suroeste_pm10                    0
sureste_ozono                    0
sureste_dioxido_de_azufre        0
sureste_dioxido_de_nitrogeno     0
sureste_monoxido_de_carbono      0
sureste_pm10                     0
dtype: int64

### Transformación de Datos

In [30]:
#Revisando tipos de datos
DF_all.dtypes

fecha                             object
hora                             float64
noroeste_ozono                   float64
noroeste_dioxido_de_azufre       float64
noroeste_dioxido_de_nitrogeno    float64
noroeste_monoxido_de_carbono     float64
noroeste_pm10                    float64
noreste_ozono                    float64
noreste_dioxido_de_azufre        float64
noreste_dioxido_de_nitrogeno     float64
noreste_monoxido_de_carbono      float64
noreste_pm10                     float64
centro_ozono                     float64
centro_dioxido_de_azufre         float64
centro_dioxido_de_nitrogeno      float64
centro_monoxido_de_carbono       float64
centro_pm10                      float64
suroeste_ozono                   float64
suroeste_dioxido_de_azufre       float64
suroeste_dioxido_de_nitrogeno    float64
suroeste_monoxido_de_carbono     float64
suroeste_pm10                    float64
sureste_ozono                    float64
sureste_dioxido_de_azufre        float64
sureste_dioxido_

Observamos que hay que cambiar la fecha a formato `datetime`. La hora no nos va a ser tan útil en análisis posteriores puesto que queremos trabajar con promedios mensuales, sin embargo vamos a transformarla a formato `int` para que no genere confusiones. Todos los demás tipos de datos son correctos.

In [31]:
#Realizando el Casting correspondiente
casting={
    "fecha":"datetime64[ns]",
    "hora":"int"
}

DF_all=DF_all.astype(casting)

In [32]:
#Comprobando que se haya realizado correctamente el casting
DF_all.dtypes

fecha                            datetime64[ns]
hora                                      int64
noroeste_ozono                          float64
noroeste_dioxido_de_azufre              float64
noroeste_dioxido_de_nitrogeno           float64
noroeste_monoxido_de_carbono            float64
noroeste_pm10                           float64
noreste_ozono                           float64
noreste_dioxido_de_azufre               float64
noreste_dioxido_de_nitrogeno            float64
noreste_monoxido_de_carbono             float64
noreste_pm10                            float64
centro_ozono                            float64
centro_dioxido_de_azufre                float64
centro_dioxido_de_nitrogeno             float64
centro_monoxido_de_carbono              float64
centro_pm10                             float64
suroeste_ozono                          float64
suroeste_dioxido_de_azufre              float64
suroeste_dioxido_de_nitrogeno           float64
suroeste_monoxido_de_carbono            

Estos datos ya están limpios y transformados al tipo correcto, así que vamos a separarlos por tipo de contaminante y a guardarlos en archivos CSV para su análisis posterior.

In [33]:
#Separando la información por contaminante
DF_save = list()
palabra = ['ozono','dioxido_de_azufre','dioxido_de_nitrogeno','monoxido_de_carbono','pm10']
for i in range(0,5):
    DF_save.append(DF_all[['fecha','hora',f'noreste_{palabra[i]}',f'suroeste_{palabra[i]}',f'noroeste_{palabra[i]}',f'sureste_{palabra[i]}',f'centro_{palabra[i]}']])

In [34]:
DF_save[0]

fecha  hora  noreste_ozono  suroeste_ozono  noroeste_ozono  \
0      2005-01-01     1           10.0            26.0            19.0   
1      2005-01-01     2            7.0            19.0            19.0   
2      2005-01-01     3            5.0            23.0            17.0   
3      2005-01-01     4            6.0            21.0            16.0   
4      2005-01-01     5            8.0            23.0            16.0   
...           ...   ...            ...             ...             ...   
140272 2020-12-31    20           14.0            25.0            25.0   
140273 2020-12-31    21           11.0            16.0            22.0   
140274 2020-12-31    22           10.0             9.0            18.0   
140275 2020-12-31    23            9.0            11.0            16.0   
140276 2020-12-31    24           11.0            16.0            16.0   

        sureste_ozono  centro_ozono  
0                13.0           4.0  
1                 9.0           4.0  
2                 9.0           4.0  
3                12.0           4.0  
4                19.0           5.0  
...               ...           ...  
140272           21.0          18.0  
140273           21.0          11.0  
140274           22.0           4.0  
140275           29.0           6.0  
140276           20.0           6.0  

[140256 rows x 7 columns]

In [35]:
#Guardamos los datos limpios y transformados en archivos CSV

#Guardando archivos separados
nombres=['O3','SO2','NO2','CO','PM10']
for i in range(0,(len(nombres))):
    DF_save[i].to_csv(f'datos_IMECA/datos_limpios_imeca/{nombres[i]}.csv',index=False)
    
#Guardando también todo el DataFrame completo
DF_all.to_csv("datos_IMECA/datos_limpios_imeca/DF_all.csv",index=False)

A continuación vamos a calcular **promedios mensuales**, agrupando todos los datos por meses. Estos son los datos que serán más útiles para nuestro análisis.

In [36]:
#Eliminando hora ya que agruparemos por meses
DF_all = DF_all.drop(['hora'], axis=1)

In [37]:
#Agrupando por mes
#Se indexa basado en tiempo, lo que permite un mejor manejo con datos temporales 
DF_all_mensual=DF_all.resample('M', on='fecha').mean()

In [38]:
DF_all_mensual

noroeste_ozono  noroeste_dioxido_de_azufre  \
fecha                                                    
2005-01-31       35.766129                   19.295699   
2005-02-28       34.148810                   12.596726   
2005-03-31       34.479839                   12.700269   
2005-04-30       38.872222                   22.769444   
2005-05-31       41.360215                   11.712366   
...                    ...                         ...   
2020-08-31       24.614595                   10.618280   
2020-09-30       25.583692                   14.509722   
2020-10-31       24.339317                   14.639785   
2020-11-30       22.882975                   10.983333   
2020-12-31       25.839057                   13.819892   

            noroeste_dioxido_de_nitrogeno  noroeste_monoxido_de_carbono  \
fecha                                                                     
2005-01-31                      22.278226                     18.849462   
2005-02-28                      20.175595                     16.983631   
2005-03-31                      19.801075                     17.076613   
2005-04-30                      22.504167                     18.434722   
2005-05-31                      19.595430                     16.596774   
...                                   ...                           ...   
2020-08-31                      12.823406                      4.209270   
2020-09-30                      14.136964                      4.452357   
2020-10-31                      15.578840                      4.613840   
2020-11-30                      15.861419                      4.801608   
2020-12-31                      16.662116                      5.424566   

            noroeste_pm10  noreste_ozono  noreste_dioxido_de_azufre  \
fecha                                                                 
2005-01-31      56.841398      34.497312                  17.502688   
2005-02-28      53.136905      34.537202                  15.058036   
2005-03-31      46.938172      37.178763                  18.165323   
2005-04-30      58.904167      43.033333                  29.266667   
2005-05-31      56.430108      45.501344                  14.360215   
...                   ...            ...                        ...   
2020-08-31      45.888441      26.763384                   7.780914   
2020-09-30      49.359722      27.409663                   9.084722   
2020-10-31      58.025538      27.760105                  13.877688   
2020-11-30      64.358333      26.872025                   8.850000   
2020-12-31      73.271505      27.893760                  11.900538   

            noreste_dioxido_de_nitrogeno  noreste_monoxido_de_carbono  \
fecha                                                                   
2005-01-31                     19.063172                    17.379032   
2005-02-28                     18.864583                    16.300595   
2005-03-31                     18.533602                    15.766129   
2005-04-30                     18.766667                    14.031944   
2005-05-31                     18.501344                    16.165323   
...                                  ...                          ...   
2020-08-31                     10.713103                     4.515812   
2020-09-30                     11.338262                     4.385783   
2020-10-31                     12.298893                     5.784629   
2020-11-30                     13.441831                     6.545849   
2020-12-31                     14.460414                     7.241508   

            noreste_pm10  ...  suroeste_ozono  suroeste_dioxido_de_azufre  \
fecha                     ...                                               
2005-01-31     76.771505  ...       42.629032                   13.668011   
2005-02-28     71.745536  ...       42.840774                    7.590774   
2005-03-31     66.627688  ...       41.713710                    9.509409   
2005-04-30     73.2013

Se observa que la agrupación se realizó correctamente, la fecha representativa es el último día del mes. A continuación guardamos este dataframe completo y separado por contaminantes.

In [39]:
#Separando la información por contaminante
DF_save = list()
palabra = ['ozono','dioxido_de_azufre','dioxido_de_nitrogeno','monoxido_de_carbono','pm10']
for i in range(0,5):
    DF_save.append(DF_all_mensual[[f'noreste_{palabra[i]}',f'suroeste_{palabra[i]}',f'noroeste_{palabra[i]}',f'sureste_{palabra[i]}',f'centro_{palabra[i]}']])

In [40]:
DF_save[0]

noreste_ozono  suroeste_ozono  noroeste_ozono  sureste_ozono  \
fecha                                                                      
2005-01-31      34.497312       42.629032       35.766129      40.125000   
2005-02-28      34.537202       42.840774       34.148810      39.645833   
2005-03-31      37.178763       41.713710       34.479839      40.633065   
2005-04-30      43.033333       46.236111       38.872222      43.913889   
2005-05-31      45.501344       50.325269       41.360215      47.463710   
...                   ...             ...             ...            ...   
2020-08-31      26.763384       32.670987       24.614595      30.297812   
2020-09-30      27.409663       31.233631       25.583692      29.125795   
2020-10-31      27.760105       33.285053       24.339317      31.169889   
2020-11-30      26.872025       31.599135       22.882975      29.184911   
2020-12-31      27.893760       32.744912       25.839057      31.688941   

            centro_ozono  
fecha                     
2005-01-31     29.528226  
2005-02-28     28.370536  
2005-03-31     28.346774  
2005-04-30     35.113889  
2005-05-31     38.028226  
...                  ...  
2020-08-31     24.317742  
2020-09-30     24.833889  
2020-10-31     25.617562  
2020-11-30     23.966065  
2020-12-31     25.019354  

[192 rows x 5 columns]

In [41]:
#Guardamos los datos limpios y transformados en archivos CSV

#Guardando archivos separados
nombres=['O3','SO2','NO2','CO','PM10']
for i in range(0,(len(nombres))):
    DF_save[i].to_csv(f'datos_IMECA/promedios_mensuales/{nombres[i]}_mensual.csv',index=True)
    
#Guardando también todo el DataFrame completo
DF_all_mensual.to_csv("datos_IMECA/promedios_mensuales/DF_all_mensual.csv",index=True)

### Resultados preliminares

A continuación trabajaremos con nuestros datos limpios y transformados para intentar responder algunas preguntas de investigación, sin un análisis tan profundo ni visualización, ya que esto se abordará en el siguiente módulo.

-   ¿En qué meses se tiene la máxima concentración de cada contaminante?

    

    
-   ¿Cuál es la zona que presenta mayor contaminación, en general?

    
-   ¿Cuál es el contaminante con más concentración por año (del 2005- 2020)?

    
-   ¿Cuál fue y cuándo se registró la concentración más baja durante 2020?

    
-   ¿Cuál fue y cuándo se registró la mayor concentración durante el periodo de confinamiento?

    
-   ¿El periodo de confinamiento coincide con una mejor calidad de aire?

    
-   ¿Cuál fue el contaminante con menores cambios en el período 2005-2020?

**¿Cuál es el promedio histórico (considerando los años 2005-2020) de concentración de cada contaminante por zona?**

In [46]:
#Promedio histórico
DF_all.set_index("fecha").mean(axis=0).round(2)

noroeste_ozono                   28.90
noroeste_dioxido_de_azufre        9.38
noroeste_dioxido_de_nitrogeno    17.48
noroeste_monoxido_de_carbono     10.09
noroeste_pm10                    55.74
noreste_ozono                    31.22
noreste_dioxido_de_azufre         7.53
noreste_dioxido_de_nitrogeno     16.09
noreste_monoxido_de_carbono       9.96
noreste_pm10                     69.31
centro_ozono                     25.97
centro_dioxido_de_azufre          6.44
centro_dioxido_de_nitrogeno      18.36
centro_monoxido_de_carbono       10.90
centro_pm10                      52.78
suroeste_ozono                   36.57
suroeste_dioxido_de_azufre        5.01
suroeste_dioxido_de_nitrogeno    15.20
suroeste_monoxido_de_carbono      8.51
suroeste_pm10                    40.04
sureste_ozono                    33.09
sureste_dioxido_de_azufre         4.68
sureste_dioxido_de_nitrogeno     15.97
sureste_monoxido_de_carbono      10.08
sureste_pm10                     51.64
dtype: float64

Los promedios históricos se observan bastante bien considerando que debajo de las 50 unidades IMECA la calidad del aire es buena.

**¿Hay algún mes de algún año en el que se haya obtenido la calificación de calidad “Extremadamente mala”?**

Recordemos que la calidad _Extremadamente mala_ corresponde a los valores mayores a 200

In [74]:
#Calidad extremadamente mala
extremadamente_mala=[]

for column in DF_all_mensual.columns:
    extremadamente_mala.append(DF_all_mensual[DF_all_mensual[column]>200])
    
extremadamente_mala=pd.concat(extremadamente_mala,axis=0)
extremadamente_mala

Empty DataFrame
Columns: [noroeste_ozono, noroeste_dioxido_de_azufre, noroeste_dioxido_de_nitrogeno, noroeste_monoxido_de_carbono, noroeste_pm10, noreste_ozono, noreste_dioxido_de_azufre, noreste_dioxido_de_nitrogeno, noreste_monoxido_de_carbono, noreste_pm10, centro_ozono, centro_dioxido_de_azufre, centro_dioxido_de_nitrogeno, centro_monoxido_de_carbono, centro_pm10, suroeste_ozono, suroeste_dioxido_de_azufre, suroeste_dioxido_de_nitrogeno, suroeste_monoxido_de_carbono, suroeste_pm10, sureste_ozono, sureste_dioxido_de_azufre, sureste_dioxido_de_nitrogeno, sureste_monoxido_de_carbono, sureste_pm10]
Index: []

[0 rows x 25 columns]

Se observa que no hay ningún registro con calidad _Extremadamente mala_ de acuerdo a los valores mensuales reportados.